cloudigrade/account/reports/__init__.py

```diff
@@ -179,3 +187,113 @@ def _calculate_instance_usage(start, end, events):
+def validate_event(event, start):
+    """
+    Ensure that the event is relevant to our time frame.
+
+    Args:
+        event: (InstanceEvent): The event object to evaluate
+        start (datetime.datetime): Start time (inclusive)
+
+    Returns:
+        bool: A boolean regarding whether or not we should inspect the event further.
+    """
+    valid_event = True
+    # if the event occurred outside of our specified period (ie. before start) we should
+    # only inspect it if it was a power on event
+    if event.occurred_at < start:
+        if event.event_type == InstanceEvent.TYPE.power_off:
+            valid_event = False
+    return valid_event
+
+def validate_account_creation(account, end):
+    """
+    Ensure that the event is relevant to our time frame.
+
+    Args:
+        account: (Account): The event object to evaluate
+        end (datetime.datetime): End time (exclusive)
+
+    Returns:
+        bool: A boolean regarding whether or not the account was created after the end
+        time.
+    """
+    if end <= account.created_at:
+        return False
+    return True
+
+
+def get_account_overview(account, start, end):
+    """
+    Generate an overview of an account over a specified amount of time.
+
+    Args:
+        start (datetime.datetime): Start time (inclusive)
+        end (datetime.datetime): End time (exclusive)
+        account (Account or AwsAccount): Account or AwsAccount object
+
+    Returns:
+        dict: An overview of the instances/images/rhel & openshift images for the specified
+        account during the specified time period."""
+    machine_images = AwsMachineImage.objects.all()
+    instances = []
+    images = []
+    rhel = []
+    openshift = []
+    cloud_type = 'unknown'
+    arn = 'unsupported'
+    # make sure that this is a supported cloud type (aws at this time)
+    if isinstance(account, AwsAccount):
+        cloud_helper = helper.get_report_helper(AWS_PROVIDER_STRING, account.aws_account_id)
+        # _get_relevant_events will return the events in between the start & end times & if
+        # no events are present during this period, it will return the last event that occurred
+        events = _get_relevant_events(start, end, cloud_helper)
+        for event in events:
+            valid_event = validate_event(event, start)
+            if valid_event:
+                instances.append(event.instance.id)
+                images.append(event.ec2_ami_id)
+                image = machine_images.filter(ec2_ami_id=event.ec2_ami_id).first()
+                if image:
+                    for tag in image.tags.all():
+                        if tag.description == 'rhel':
+                            rhel.append(image)
+                        if tag.description == 'openshift':
+                            openshift.append(image)
+        # set aws account info & grab the totals
+        cloud_type = AWS_PROVIDER_STRING
+        arn = account.account_arn
+        id = account.aws_account_id
```

Comment:
`id` is a built-in function in Python, and I'm surprised `flake8` didn't complain about overwriting it. Regardless, we should name this variable something else so we don't overwrite `id`.

Reply:
Good catch! Weird flake8 didn't catch this 🤷‍♀️